### PR TITLE
Add internationalization to payment method clean name

### LINKED ISSUE #6535: Missing Translation: Payment Method types
## Description

Translation keys for the different payment method types are missing:
- Cash/EFT/etc (Payments for which automatic validation is not required)
- Paypay Express
- Stripe
- Stripe SCA
- Pinpayments
- Mastercard Internet Gateway Service (MIGS) <-- though I believe we are removing this one

While we are deprecating and removing Stripe (Stripe Connect) payment method it would be useful to be able to rename it. 

## Expected Behavior

These fields can all be translated.

## Actual Behaviour

They can't

## Steps to Reproduce

1. Go to create a new payment method
2. The payment method types are all hard coded in English

## Animated Gif/Screenshot

Screenshot from 2020-12-15 10-44-05

## Severity

bug-s3: a feature is broken but there is a workaround

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by PauloVilarinho**
#### What? Why?

Closes #6535 

This adds internationalization to the payment method model clean name method, fixing the internationalzation of the new payment method at the admin panel

#### What should we test?

Test create new payment method at admin panel

#### Release notes

Internationalization added at creating new payment method at admin panel for the payment method type

Changelog Category: User facing changes

### CODE REVIEW

File: app/models/spree/payment_method.rb [hunk 1]
```diff
@@ -110,23 +110,9 @@ def has_distributor?(distributor)
...
-        "Stripe SCA"
-      when "Spree::Gateway::PayPalExpress"
-        "PayPal Express"
-      else
-        i = name.rindex('::') + 2
-        name[i..-1]
-      end
+      i = name.rindex('::') + 2
+      i18n_name = "spree.admin.payment_methods.providers." + name[i..-1].downcase
+      I18n.t(i18n_name)
```
Comment (Matt-Yorkley) [inline app/models/spree/payment_method.rb:115]: I'm not a big fan of these single-letter variables and arbitrary numbers. Could we use something a bit more readable here? Maybe something like: 
```ruby
I18n_key = "spree.admin.payment_methods.providers." + name.demodulize.downcase
I18n.t(I18n_key)
```
https://apidock.com/rails/v4.2.1/String/demodulize

Comment (PauloVilarinho) [reply app/models/spree/payment_method.rb:115]: Yeah, that seems a lot more readable. I'm changing this now

Comment (PauloVilarinho) [reply app/models/spree/payment_method.rb]: Done @Matt-Yorkley

**Review by Matt-Yorkley — APPROVED** [review]
LGTM, thanks @PauloVilarinho :heart:

File: app/models/spree/payment_method.rb (hunk 1 above)
Comment (Matt-Yorkley) [reply app/models/spree/payment_method.rb]: That looks much nicer now, and the specs are still passing :ok_hand:

Comment (andrewpbrett) [reply app/models/spree/payment_method.rb]: Agree, that's a nice cleanup.

**Review by andrewpbrett — APPROVED** [review]
Thanks @PauloVilarinho !

**Comment by andrewpbrett** [comment]
@PauloVilarinho I moved this to the Test Ready column on our ZenHub. It may be a little while before it gets merged deployed since not as many people are around in the next couple weeks, but it'll get in there soon! And we can take a look at the other PR you sent in the meantime. Thanks again! 🎉

**Comment by filipefurtad0** [comment]
Olá @PauloVilarinho ,
Welcome!

Thanks for fixing this. I see you added the keys for this in `en.yml`, the page renders correctly, so all should work once the translations are updated on Transifex:

image

The arrow above spots another missing translation - unrelated to this PR.

Looks good, moving to ready to go!
Thank you :tada:

**Comment by RachL** [comment]
@Matt-Yorkley  @andrewpbrett I see this has been included in the release deployed on Tuesday yet I don't see the strings in Transifex. Can you confirm which strings I should be looking for?

**Comment by sigmundpetersen** [comment]
I see the auto update of en.yml on Transifex has been turned off since Dec 9th.
I guess this is done while we're trying to find the best process for translation between releases.
But it means there has not been pushed any strings to Transifex since that date ⚠️

**Comment by andrewpbrett** [comment]
Ohhh good catch @sigmundpetersen. I was just opening transifex to check the keys; see also this discussion: https://github.com/openfoodfoundation/openfoodnetwork/pull/6508#discussion_r556991512

**Comment by sauloperez** [comment]
would you mind turning it on meanwhile @sigmundpetersen ?

**Comment by andrewpbrett** [comment]
Is there an admin login to Transifex? I don't see the button to turn the updating back on, and I don't see an admin login in bitwarden. Or maybe someone could make me an admin in Transifex :)

**Comment by sigmundpetersen** [comment]
I added back the url, will wait and see if it updates.
I added you as admin on Transifex @andrewpbrett , you should see the button now 🙂

**Comment by sigmundpetersen** [comment]
>  the check happens 4 times a day

I guess we will have to wait a while 🙈 
Or we can push an update with the command line client
https://docs.transifex.com/projects/updating-content/

**Comment by andrewpbrett** [comment]
Okay I just ran `tx push -s` with the latest master and I see the string `spree.admin.payment_methods.providers.stripesca`, for example, in Transifex.

**Comment by RachL** [comment]
Along with 24 other new strings 😅 yay! Thanks guys

**Pull request merged**